# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/threads/TaskQueue.cpp
# Commit: 33ae2aabdf22
# Full Hash: 33ae2aabdf225a72dfceb09424e4e43ae9430e69
# Author: Atila Butkovits <abutkovits@mozilla.com>
# Date: 2025-06-12 16:31:09
# Regressor Bug: 1968812
# File Overlap Count: 1
# Description:
#   Revert "Bug 1968812 - Remove the CallWorker thread and use TaskQueue::Observer instead. r=bwc" for causing failures complaining about mutexLock.
#   
#   This reverts commit d53f26b530b4ad59aebe314c55a8af2c62a80fc3.
#   
#   Revert "Bug 1968812 - Copy WebrtcTaskQueueWrapper.h to WebrtcTaskQueueWrapper.cpp. r=bwc"
# ==============================================================================

diff -r ca232b0163c7 -r 33ae2aabdf22 xpcom/threads/TaskQueue.cpp
--- a/xpcom/threads/TaskQueue.cpp	Wed Jun 11 22:34:11 2025 +0300
+++ b/xpcom/threads/TaskQueue.cpp	Wed Jun 11 23:24:24 2025 +0300
@@ -222,7 +222,6 @@
     // Disconnect from our target as we won't try to dispatch any more events.
     mTrackerEntry = nullptr;
     mTarget = nullptr;
-    mObserver = nullptr;
   }
 }
 
@@ -236,15 +235,8 @@
   return in;
 }
 
-void TaskQueue::SetObserver(Observer* aObserver) {
-  MonitorAutoLock mon(mQueueMonitor);
-  MOZ_ASSERT_IF(aObserver, !mObserver);
-  mObserver = std::move(aObserver);
-}
-
 nsresult TaskQueue::Runner::Run() {
   TaskStruct event;
-  RefPtr<Observer> observer;
   {
     MonitorAutoLock mon(mQueue->mQueueMonitor);
     MOZ_ASSERT(mQueue->mIsRunning);
@@ -255,7 +247,6 @@
       return NS_OK;
     }
     event = mQueue->mTasks.Pop();
-    observer = mQueue->mObserver;
   }
   MOZ_ASSERT(event.event);
 
@@ -265,7 +256,7 @@
   // designed to be threadsafe, it will be, provided we're only calling it
   // in this task queue.
   {
-    AutoTaskGuard g(mQueue, observer);
+    AutoTaskGuard g(mQueue);
     {
       LogRunnable::Run log(event.event);
 